src/main/java/duke/utils/Parser.java

```diff
@@ -0,0 +1,90 @@
+package duke.utils;
+
+import duke.task.Deadlines;
+import duke.task.Events;
+import duke.task.Task;
+import duke.task.Todos;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.ArrayList;
+import java.util.List;
+
+public class Parser {
+
+    public static Date stringToDate(String text) {
+        Date date = null;
+        SimpleDateFormat formatter = new SimpleDateFormat(Constants.DATE_FORMAT_1);
+        try {
+            date = formatter.parse(text);
+        } catch (ParseException e) {
+            e.printStackTrace();
+        }
+        return date;
+    }
+
+    public static boolean checkDateFormat(String text) {
+        String[] tokens = text.split("-");
+        if (tokens[0].length() != 4) {
+            return false;
+        }
+        if (tokens[1].length() != 2 || tokens[1].compareTo("00") <= 0 || tokens[1].compareTo("12") > 0) {
+            return false;
+        }
+        if (tokens[2].length() != 2 || tokens[2].compareTo("00") <= 0 || tokens[2].compareTo("31") > 0) {
+            return false;
+        }
+        return true;
+    }
+
+    public static String dateToString(Date date) {
+        SimpleDateFormat formatter = new SimpleDateFormat(Constants.DATE_FORMAT_2);
+        String text = formatter.format(date);
+        return text;
+    }
+
+    public static String tasksToStorage(List<Task> tasks) {
+        String data = "";
+        SimpleDateFormat formatter = new SimpleDateFormat(Constants.DATE_FORMAT_1);
+        for (int i = 0; i < tasks.size(); i++) {
+            Task task = tasks.get(i);
+            String doneString = task.isDone()? "1" : "0";
```

Comment:
Perhaps you might want to consider doing this in the individual 'ToDo', 'Deadline' and 'Event' class or even in the 'Task' class to make your code look neater.